spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -132,16 +137,36 @@ public PubSubTemplate setMessageConverter(PubSubMessageConverter messageConverte
 	}
 
 	@Override
+	@Deprecated
 	public Subscriber subscribe(String subscription, MessageReceiver messageReceiver) {
 		return this.pubSubSubscriberTemplate.subscribe(subscription, messageReceiver);
 	}
 
+	@Override
+	public Subscriber subscribe(String subscription, Consumer<BasicAcknowledgeablePubsubMessage> messageConsumer) {
+		return getPubSubSubscriberTemplate().subscribe(subscription, messageConsumer);
+	}
+
+	@Override
+	public <T> Subscriber subscribeAndConvert(String subscription,
+			Consumer<ConvertedBasicAcknowledgeablePubsubMessage<T>> messageConsumer,
+			Class<T> payloadType) {
+		return this.pubSubSubscriberTemplate.subscribeAndConvert(subscription, messageConsumer, payloadType);
```

Comment:
Why inconsistency? In the previous method you use `getPubSubSubscriberTemplate()`, but here it is a `this.pubSubSubscriberTemplate`...

Reply:
the previous one is wrong